pkg/controllers/networking/node_controller.go

```diff
@@ -69,12 +69,22 @@ func (r *NodeReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
-	var underlayAttached, overlayAttached bool
-	if underlayAttached, overlayAttached, err = utils.DetectNetworkAttachmentOfNode(r, node); err != nil {
+	var underlayAttached, overlayAttached, bgpAttached bool
+	if underlayAttached, overlayAttached, bgpAttached, err = utils.DetectNetworkAttachmentOfNode(r, node); err != nil {
```

Comment:
I'm afraid I still want bgp attachment detection is an independent function.

Reply:
As you wish.